packages/next/src/client/dev/hot-reloader/app/hot-reloader-app.tsx

```diff
@@ -456,96 +458,40 @@ function processMessage(
 }
 
 export default function HotReload({
-  assetPrefix,
   children,
   globalError,
+  webSocket,
+  staticIndicatorState,
 }: {
-  assetPrefix: string
   children: ReactNode
   globalError: GlobalErrorState
+  webSocket: WebSocket | undefined
+  staticIndicatorState: StaticIndicatorState | undefined
 }) {
   useErrorHandler(dispatcher.onUnhandledError, dispatcher.onUnhandledRejection)
-
-  const webSocketRef = useWebSocket(assetPrefix)
-
-  useWebSocketPing(webSocketRef)
-  const sendMessage = useSendMessage(webSocketRef)
-  useForwardConsoleLog(webSocketRef)
-  const processTurbopackMessage = useTurbopack(sendMessage, (err) =>
-    performFullReload(err, sendMessage)
-  )
-
-  const router = useRouter()
+  useWebSocketPing(webSocket)
 
   // We don't want access of the pathname for the dev tools to trigger a dynamic
   // access (as the dev overlay will never be present in production).
   const pathname = useUntrackedPathname()
-  const appIsrManifestRef = useRef<Record<string, false | number>>({})
-  const pathnameRef = useRef(pathname)
 
   if (process.env.__NEXT_DEV_INDICATOR) {
     // this conditional is only for dead-code elimination which
     // isn't a runtime conditional only build-time so ignore hooks rule
     // eslint-disable-next-line react-hooks/rules-of-hooks
     useEffect(() => {
-      pathnameRef.current = pathname
-
-      const appIsrManifest = appIsrManifestRef.current
-
-      if (appIsrManifest) {
-        if (pathname && pathname in appIsrManifest) {
-          try {
-            dispatcher.onStaticIndicator(true)
-          } catch (reason) {
-            let message = ''
-
-            if (reason instanceof DOMException) {
-              // Most likely a SecurityError, because of an unavailable localStorage
-              message = reason.stack ?? reason.message
-            } else if (reason instanceof Error) {
-              message = 'Error: ' + reason.message + '\n' + (reason.stack ?? '')
-            } else {
-              message = 'Unexpected Exception: ' + reason
-            }
-
-            console.warn('[HMR] ' + message)
-          }
+      if (staticIndicatorState) {
```

Comment:
`staticIndicatorState` being `undefined` here would be a bug. We should surface that potential bug. This is a dev-only file and in dev, this value is always defined.